File: java/src/test/java/com/testexercise/ebayframework/WebDriverFactory.java
package com.testexercise.ebayframework;

import java.util.*;

import org.openqa.selenium.*;
import org.openqa.selenium.chrome.*;
import org.openqa.selenium.firefox.*;

public class WebDriverFactory
{
    public static WebDriver getDriver(
            WebBrowserType browser,
            String driverPath)
    {
        switch (browser)
        {
            case CHROME:
                return getChromeDriver(driverPath);
            case FIREFOX:
                return getFirefoxDriver(driverPath);
            default:
                throw new IllegalArgumentException("Unknown browser type");
        }
    }
    
    public static WebDriver getChromeDriver(String path)
    {
        System.setProperty("webdriver.chrome.driver", path);
       
        ChromeOptions options = new ChromeOptions();
        options.addArguments("start-maximized", "disable-extensions");
        
        Map<String, Object> prefs = new HashMap<String, Object>();
        prefs.put("profile.default_content_settings.popups", 0);
        prefs.put("spellcheck.dictionary", "");
        prefs.put("credentials_enable_service", false);
        prefs.put("profile.password_manager_enabled", false);
        
        options.setExperimentalOption("prefs", prefs);
        
        return new ChromeDriver(options);
    }
    
    public static WebDriver getFirefoxDriver(String path)
    {
        System.setProperty("webdriver.gecko.driver", path);
        
        return new FirefoxDriver();
    }
}

File: net/TestExerciseEbay/EbayPOMFramework/WebDriverFactory.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Firefox;

namespace TestExercise.EbayPOMFramework
{
    public class WebDriverFactory
    {
        public static IWebDriver GetDriver(
                WebBrowserType browser,
                String driverPath)
        {
            switch (browser)
            {
                case WebBrowserType.CHROME:
                    return GetChromeDriver(driverPath);
                case WebBrowserType.FIREFOX:
                    return GetFirefoxDriver(driverPath);
                default:
                    throw new ArgumentException("Unknown browser type");
            }
        }

        public static IWebDriver GetChromeDriver(String path)
        {
            ChromeOptions options = new ChromeOptions();

            options.AddArguments("start-maximized", "disable-extensions");

            options.AddLocalStatePreference("profile.default_content_settings.popups", 0);
            options.AddLocalStatePreference("spellcheck.dictionary", "");
            options.AddLocalStatePreference("credentials_enable_service", false);
            options.AddLocalStatePreference("profile.password_manager_enabled", false);

            if (string.IsNullOrWhiteSpace(path))
            {
                return new ChromeDriver(options);
            }
            else
            {
                return new ChromeDriver(path, options);
            }
        }

        public static IWebDriver GetFirefoxDriver(String path)
        {
            if (string.IsNullOrWhiteSpace(path))
            {
                return new FirefoxDriver();
            }
            else
            {
                return new FirefoxDriver(path);
            }
        }
    }
}
